### ISSUE TITLE
[Feature Request]: expose prove_shares and data_root_inclusion_proof in node API

### ISSUE THREAD

**Issue by liamsi** [celestiaorg/celestia-node#3364]
### Implementation ideas

These two methods are important for integrations/blobstream:
https://github.com/celestiaorg/celestia-app/issues/3421#issuecomment-2091191736

First step could be to expose these two methods in app and simply proxy them in the node API. Then, they could be moved to node entirely.

**Comment by Wondertan** [comment]
Eventually, proxying will be replaced by the work @rach-id is doing which will compute these proofs  solely relying on DA node functionality.

**Comment by liamsi** [comment]
Yes, agreed. that's why the OP mentions two steps: 1) proxying and 2) fold/move that functionality entirely into node.

**Comment by renaynay** [comment]
This will not make it to v0.14.0, but may be a part of v0.15.0

**Comment by renaynay** [comment]
@liamsi @Bidon15 

I don't think this feature will be ready in time to land in v0.14.0 but that means it would have to be a part of v0.15.0 which may be a couple months out. 

Please confirm that this is alright.

**Comment by Bidon15** [comment]
> @liamsi @Bidon15
> 
> I don't think this feature will be ready in time to land in v0.14.0 but that means it would have to be a part of v0.15.0 which may be a couple months out.
> 
> Please confirm that this is alright.

Totally fine for me and thanks for the heads-up 🙏 👑

**Comment by liamsi** [comment]
If it is fine for Viet, than it is fine for me. 
>  it would have to be a part of v0.15.0 which may be a couple months out.

Ideally, we would have an RC including these endpoints that can already be used earlier.

**Comment by renaynay** [comment]
@liamsi an RC will always come before the release :)

**Comment by liamsi** [comment]
I know. Is a RC with the above also months away though?

**Comment by damiannolan** [comment]
related: https://github.com/celestiaorg/celestia-node/issues/3473
